Arduino DFA Simulation/serial_monitor_to_file.py

```diff
@@ -51,16 +55,18 @@ def save_to_file(traces: list) -> None:
 serial_instance.baudrate = baud_rate
 serial_instance.port = active_port
 serial_instance.open()
+time.sleep(2)
 
 while 1:
     # message is read from serial
     arduino_data = serial_instance.readline()
-
+    print(arduino_data.decode().strip())
     # add read message to list
-    received_messages.append(arduino_data.decode().rstrip())
+    if 'STOP' not in str(arduino_data):
+        received_messages.append(arduino_data.decode().strip())
 
     # if we read STOP, we stream all data to file
-    if 'STOP' in str(arduino_data.decode()):
+    if 'STOP' in str(arduino_data):
         save_to_file(received_messages)
```

Comment:
Hvorfor bruger i ikke længere decode? readline fra serial_instance encoder i utf-8 format, så derfor bruges decode til at få den originale string tilbage.

Reply:
Fordi in, finder det aligevel, python tjekker bare, efter string, Så det er bare extra computation

Reply:
Det er sat tilbage igen, da hvis vi sende b ville decode self fuckke